webapp/src/index.tsx

```diff
@@ -16,13 +16,40 @@ export default class Plugin {
     enforceConnectedAccountId = '';
     // eslint-disable-next-line @typescript-eslint/no-unused-vars, @typescript-eslint/no-empty-function
     public async initialize(registry: PluginRegistry, store: Store<GlobalState, Action<Record<string, unknown>>>) {
-        Client.setServerRoute(getServerRoute(store.getState()));
+        const serverRoute = getServerRoute(store.getState());
+        Client.setServerRoute(serverRoute);
 
         // @see https://developers.mattermost.com/extend/plugins/webapp/reference/
         this.enforceConnectedAccountId = registry.registerRootComponent(EnforceConnectedAccountModal);
 
         registry.registerAdminConsoleCustomSetting('appManifestDownload', MSTeamsAppManifestSetting);
         registry.registerAdminConsoleCustomSetting('ConnectedUsersReportDownload', ListConnectedUsers);
+        registry.registerUserSettings?.({
+            id: manifest.id,
+            icon: `${serverRoute}/plugins/${manifest.id}/public/msteams-sync-icon.svg`,
+            uiName: 'MS Teams Sync',
```

Comment:
There are two ways to handle this. Either we add here `manifest.name`, and let it be the same as the manifest, or we change it in the client side, so the client gets the manifest and use the name there.

The first option still leaves liberty to the plugin to use a different team if they desire, but at the same time reduces the maintenance cost because we would need to change the name only in one place.

The second option removes the option of changing the name from the plugin. It will always use the one in the manifest. I don't dislike it, but I thought we wanted to give this piece of freedom to the plugin. Worst case scenario, it can be added later.

Regarding the 32 char limit, that one is about the id and creating the preference category. The UI limit is not really about number of characters, but space (it is affected by many variables, like the width of each letter (a `W` is wider than a `i`)). I don't expect often the plugin developer to have an overcomplicated name they want to simplify in the settings. 0/5 on this.

Reply:
Thanks for the explanation here @larkox! I see you went with your first option here which still leaves the flexibility for plugin authors to use anything else if they would like. 👍